python/GafferSceneTest/IECoreScenePreviewTest/CapturingRendererTest.py

```diff
@@ -165,5 +168,390 @@ def testDeformingObject( self ) :
 		self.assertEqual( c.capturedSamples(), [ sphere1, sphere2 ] )
 		self.assertEqual( c.capturedSampleTimes(), [ 1, 2 ] )
 
+	class TestProcedural( GafferScene.Private.IECoreScenePreview.Procedural ) :
+
+		def __init__( self, n ) :
+
+			GafferScene.Private.IECoreScenePreview.Procedural.__init__( self )
+			self.n = n
+
+		def render( self, renderer ) :
+
+			attributes = renderer.attributes( IECore.CompoundObject( { "a" : IECore.FloatData( 12 ) } ) )
+			self.o = renderer.object( "a", IECoreScene.SpherePrimitive( 42 ), attributes )
+			self.o.transform( imath.M44f().translate( imath.V3f( 3, 0, 0 ) ) )
```

Comment:
Why store the ObjectInterface as a member? It's not expected that procedurals would retain ownership of the objects they create.